recipes/llvm-core/all/conanfile.py

```diff
@@ -0,0 +1,266 @@
+        cmake.definitions['LLVM_ENABLE_LIBPFM'] = False
+        cmake.definitions['LLVM_ENABLE_LIBEDIT'] = False
+
+        cmake.definitions['LLVM_ENABLE_ZLIB'] = \
+            self.options.get_safe('with_zlib', False)
+        cmake.definitions['LLVM_ENABLE_LIBXML2'] = \
+            self.options.get_safe('with_xml2', False)
+        cmake.definitions['LLVM_ENABLE_FFI'] = \
+            self.options.get_safe('with_ffi', False)
+        if self.options.get_safe('with_ffi', False):
+            cmake.definitions['FFI_INCLUDE_DIR'] = \
+                self.deps_cpp_info['libffi'].include_paths[0]
+            cmake.definitions['FFI_LIBRARY_DIR'] = \
+                self.deps_cpp_info['libffi'].lib_paths[0]
+        return cmake
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+            del self.options.components
+            del self.options.with_zlib
+            del self.options.with_xml2
+            del self.options.with_ffi
+
+    def requirements(self):
+        if self.options.get_safe('with_zlib', False):
+            self.requires('zlib/1.2.11')
+        if self.options.get_safe('with_xml2', False):
+            self.requires('libxml2/2.9.10')
+        if self.options.get_safe('with_ffi', False):
+            self.requires('libffi/3.3')
+
+    def configure(self):
+        if self.settings.os == 'Windows' and self.options.shared:
+            raise ConanInvalidConfiguration('shared lib not supported')
+        if self.options.exceptions and not self.options.rtti:
+            raise ConanInvalidConfiguration('exceptions require rtti support')
+
+    def source(self):
+        tools.get(**self.conan_data['sources'][self.version])
+        source_path = Path(f'llvm-{self.version}.src')
+        source_path.rename(self._source_subfolder)
+
+        if self.version in self.conan_data['patches']:
+            for patch in self.conan_data['patches'][self.version]:
+                tools.patch(**patch)
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        self.copy('LICENSE.TXT', dst='licenses', src=self._source_subfolder)
+        package_path = Path(self.package_folder)
+
+        cmake = self._configure_cmake()
+        cmake.install()
+
+        if not self.options.shared:
+            lib_regex = re.compile(
+                r'add_library\((\w+?)\s.*?\)'
+                r'(?:(?:#|\w|\s|\()+?'
+                r'INTERFACE_LINK_LIBRARIES\s\"((?:;|:|/|\.|\w|\s|-|\(|\))+?)\"'
+                r'(?:.|\n)*?\))?'
+            )
+            exports_file = 'LLVMExports.cmake'
+            exports_path = package_path.joinpath('lib', 'cmake', 'llvm')
+            exports_path = exports_path.joinpath(exports_file)
+
+            exports = tools.load(str(exports_path.resolve()))
+            exports = exports.replace(r'\$<LINK_ONLY:', '')
+            exports = exports.replace(r'>', '')
```

Comment:
I think it's a very bad way to do because it's hard to maintain. However, there is no good way to do this at the moment. Maybe by using the cmake file API, we can extract the targets and their dependencies.

Reply:
Oh, that's for sure. I was hopping that someone could come up with a clever solution for this. And the problems don't even end up there...

If you take a look below, among the libraries there are ones that use full paths instead of their name, another ones comes prepended by *-l* and, in **Windows**, a couple are linked using *delayload* (which is not true, as they're specified to be linked in the usual way too for that same target).

```python
components[lib] = []
for dep in deps.split(';') if deps is not None else []:
    if Path(dep).exists():
        dep = Path(dep).stem.replace('lib', '')
    elif dep.startswith('-delayload:'):
        continue
    components[lib].append(dep.replace('-l', ''))
```

I end up using regular expressions as it was a quick solution at the moment and it is not expected for the exports file to change substantially in format, but any other alternative to this is welcome.

Reply:
I think I'm resolving this conversation as it no longer applies.